web/src/main/java/com/microsoft/applicationinsights/web/extensibility/modules/WebRequestTrackingTelemetryModule.java

```diff
@@ -94,7 +120,13 @@ public void onBeginRequest(ServletRequest req, ServletResponse res) {
 
             // Look for cross-component correlation headers and resolve correlation ID's
             HttpServletResponse response = (HttpServletResponse) res;
-            TelemetryCorrelationUtils.resolveCorrelation(request, response, telemetry);
+            if (isW3CEnabled) {
+                TraceContextCorrelation.resolveCorrelation(request, response, telemetry);
```

Comment:
we should be backward compatible and if w3c headers are not present, check for legacy headers